lib/logstash/inputs/gelf.rb

```diff
@@ -157,7 +159,7 @@ def self.from_json_parse(json)
   # legacy_parse uses the LogStash::Json class to deserialize json
   def self.legacy_parse(json)
     o = LogStash::Json.load(json)
-    LogStash::Event.new(o) if o
+    LogStash::Event.new(o)
```

Comment:
yes it can https://github.com/elastic/logstash/blob/4a3cbb48685a991726518ee243e50d961c3888be/logstash-core/spec/logstash/json_spec.rb#L114-L124